### PR TITLE
Minor Fix + Toy Example

### PR DESCRIPTION
**Pull request by ericspod**
I've put a fix in residual_units.py to account for adding tuple and list objects and found a few more instances of incorrect assert syntax.

The toy example is based on that provided by https://github.com/jakeret/tf_unet which creates a noisy image with circles and a mask. This I feel is a more useful example than the MR brain one for getting used to the programming interface since it doesn't hide what the format of input data is. 

However, it doesn't produce acceptable results yet so I'm assuming something is wrong in my implementation. Using the same worksheet with tf_unet's definition of UNet produces good results very similar to their own toy example but I'm not getting the same results out of DLTK. If this can be integrated and fixed I think it would help a lot of people, myself included, in getting going with DLTK.

### CODE REVIEW

**Comment by pawni** [comment]
Thanks for putting this together! The problem with the toy example is that our dice loss implementation is defined as multiclass dice and currently doesn't handle single class cases. Because of that the optimisation is incorrect. Furthermore, you are plotting the logits rather than the actual predictions.

File: examples/Toy_segmentation/dltk_test.ipynb [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
...alUAEC6HL1eDz6fT4z328nZ3HLkaj2B\nlQDQ7Xbhdruxb9++1zUKPJXdKxYEZ+vG4MGDB2XBgyPh7Zue7FCRLcURb7lclsxKFt/NZhPT6RR+\nv18yK+kPXFhYQKPREK8WA7w5UrTZbDh06JAY6P1+v2z6sejnec8bjc3NTWSzWRlNe71etFot6cgC\nWx227X7FueY62dq3bx+++MUvIplMvvyT5zqpeuCBB/DEE0/M+jBOK820c5XJZDAYDBCPx2G324WH\nQ3p4oVCQC8xgMEC5XJaxGi9Ew+FQttwY7UE/FDtTwWAQVqsVk8kElUpFNv04arPb7YjFYgiFQgAg\nY8PxeIxCoSBFDIGNHFcyY43bjuwAdDoduajxbp8XlsFgIPEzHIkOBgNsbm7Kc0wmk3S4thPiGQxN\nfxgAKTgJaex2u5hMJpKByKgfBkgrioJ6vS4bjVarFXv37sU111zzuj1WjGQ5FeLI9GzV3XffjWPH\njgkGg0VKNBqVtAAuR3BxgwWRyWRCOBxGIpEQPEkgEJBilx2x6XQq3VsWwRzl+f1+jEYj1Go1Cfsm\n32o8HqPZbEr3jD+LcrmMcrmMzc1NWY7I5/NwOBwoFAqSKhCJRDAcDrG0tDTrt3mus1T79u3DF77w\nBSwuLs76UN6Q+u53v4sf/ehHsz6M00ozLa5cLheGwyGazSZUVUWhUBBu1GAwkO5Sr9dDpVKBz+cT\nkjuLjVgsBlVV4Xa7USgUpOsDQOI4SDcnMJRbTd1uF81mE61WSwCMZPXQf+X1eiUvkMZ3jtz4OEds\nzOdjN85sNqPdbqPRaMhFT9d1dDodyTBsNptIp9NyweJGocPhkIgZxp20220Jn+ZYsd1uC/6h1+sh\nm82KaZ/r/Hy/Op2ObEvyOFwuF970pjfhfe97QW/vqxbZYHO9OrFTVa/XpWMLbHUD6fUj8NVkMkHT\nNKTTaZjNZmGS9Xo9bGxsyIiWm4I0my8uLkqBRZq/w+GAz+eD1+uF2+1Gv99HKBRCNBqVMR7jbAjV\n5c3NeDwWWKmqqlhaWpLPq67raLVaWF9fh81mk23CueY6FfqTP/mTefE+12mlmf62Y6gyAOmwEPTJ\nURxHE+PxGO12G7lcDpPJBNVqVUKGiXFg0dJut9HtdsWLxK2qVquFVquFTCYDq9WKUqkkYxH+2el
```
Comment (pawni) [inline examples/Toy_segmentation/dltk_test.ipynb]: You either need to run `net(x)['y_'].eval(..)` or plot `np.squeeze(np.argmax(test[0]))`

Comment (pawni) [inline examples/Toy_segmentation/dltk_test.ipynb]: Our dice currently only handles multi class cases - I managed to get the example working by changing everything to 2 classes. However, then the training data would be somewhat incorrect.

**Comment by ericspod** [comment]
Thanks for the responses. I've updated the example and it works a lot better now with two classes treated as inverses of one another, a dice definition with Jaccard formulation, and using the y_ value from the net. I think using `num_classes=1` as you suggest doesn't function correctly so this was the solution I've used elsewhere.

**Comment by pawni** [comment]
Looks good. I meant that we would need to implement a fix for the special case of `num_classes=1` within the `dice_loss` and network classes. But this multi-class approach works for now.
